### PR TITLE
Add withDelete and fix unwind parameters

### PR DESCRIPTION
**Pull request by angrykoala**
# Description

Adds `.delete` and `.detachDelete` to some clauses and removes Unwind invalid parameters

## Complexity

Complexity: Low

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 35 | 41 | 1134 |
|  aggregations.NestedAggregation | 14551 | 14551 | 83 | 97 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 64 | 65 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3239 | 3063 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3111 | 3125 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 42 | 48 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 79 | 50 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070331 | 4851 | 5014 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 143 | 181 | 600 |
|  connect.createAndConnect | 14424 | 14424 | 152 | 237 | 3003 |
|  connections.Connection | 14082 | 14082 | 55 | 76 | 2174 |
|  connections.NestedConnection | 41462 | 41462 | 101 | 183 | 4516 |
|  connections.ConnectionWithSort | 2282 | 2282 | 59 | 76 | 1040 |
|  connections.ConnectionWithSortAndCypher | 2282 | 2282 | 188 | 306 | 1040 |
|  create.SimpleMutation | 7 | 7 | 59 | 62 | 1 |
|  cypher-directive.TopLevelCypherDirective | 116092 | 116092 | 573 | 726 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 147611 | 147609 | 152 | 207 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 274 | 308 | 1040 |
|  delete.NestedDeleteInUpdate | 22200 | 22200 | 136 | 188 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 63 | 42 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 73 | 83 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 3450 | 3450 | 23 | 83 | 1134 |
|  2925.NestedSingleRelationshipRequiredFilter | 9383 | 9383 | 48 | 51 | 1040 |
|  query.SimpleQuery | 15160 | 15160 | 43 | 44 | 2174 |
|  query.QueryWhere | 9699 | 9715 | 35 | 48 | 2163 |
|  query.SimpleQueryWithNestedWhere | 9877 | 9893 | 51 | 60 | 2163 |
|  query.Nested | 10092037 | 10092037 | 10813 | 10450 | 2008534 |
|  query.NestedWithFilter | 40401 | 40401 | 98 | 131 | 4511 |
|  query.OrFilterOnRelationships | 41402 | 43618 | 150 | 173 | 1228 |
|  query.OrFilterOnRelationshipsAndNested | 35118 | 37294 | 196 | 240 | 1228 |
|  query.QueryWithNestedIn | 13308 | 13699 | 67 | 59 | 1189 |
|  query.NestedConnectionWhere | 9834 | 9834 | 50 | 60 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10024 | 10084 | 75 | 102 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19242 | 19314 | 149 | 148 | 1630 |
|  query.NestedWithRelationshipSingleFilters | 3024400 | 3024400 | 623 | 726 | 1003942 |
|  query.Fulltext | 96 | 96 | 25 | 35 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 43 | 60 | 84 |
|  sorting.SortMultipleTypes | 2513 | 2513 | 63 | 70 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1450 | 1507 | 161 | 237 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 47 | 62 | 2174 |
|  sorting.SortDeeplyNestedFields | 42196 | 42196 | 106 | 96 | 4516 |
|  sorting.SortWithTopLevelCypher | 2119 | 2119 | 146 | 190 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 75 | 68 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 105 | 76 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 387 | 324 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 285 | 307 | 33033 |
|  update.NestedUpdate | 2119 | 2119 | 63 | 87 | 1040 |

</details>
        
Old Schema Generation: 40.148s
Schema Generation: 39.619s
Old Subgraph Schema Generation: 56.277s
Subgraph Schema Generation: 53.377s

File: packages/cypher-builder/src/clauses/sub-clauses/Delete.ts [hunk 1]
```diff
@@ -19,10 +19,9 @@
 
 import { CypherASTNode } from "../../CypherASTNode";
 import type { CypherEnvironment } from "../../Environment";
-import type { NodeRef } from "../../references/NodeRef";
-import type { RelationshipRef } from "../../references/RelationshipRef";
+import type { Variable } from "../../references/Variable";
 
-export type DeleteInput = Array<NodeRef | RelationshipRef>;
+export type DeleteInput = Array<Variable>;
```
Comment (MacondoExpress) [inline packages/cypher-builder/src/clauses/sub-clauses/Delete.ts:24]: I believe this should be:
```suggestion
export type DeleteInput = Array<NodeRef | RelationshipRef | Variable>;
```

**Review by MacondoExpress — COMMENTED** [review]
LGTM,  I would change the type of DeleteInput, IMO is true that Cypher.Node/Cypher.Relationship can be coerced to Cypher.Variable but this is true only because of the Cypher.Variable class has not diverged yet from the Reference class.

File: packages/cypher-builder/src/clauses/sub-clauses/Delete.ts (hunk 1 above)
Comment (angrykoala) [reply packages/cypher-builder/src/clauses/sub-clauses/Delete.ts]: Good catch

**Review by MacondoExpress — APPROVED** [review]

**Review by mjfwebb — APPROVED** [review]

**Pull request merged**